aerleon/lib/proxmox.py

```diff
@@ -0,0 +1,658 @@
+                                self.term.logging,
+                                self.term.option,
+                            )
+                        )
+
+        return '\n'.join(ret_str)
+
+    def _Format(
+        self,
+        protocol: Union[str, None],
+        direction: str,
+        action: str,
+        source_addresses: List[Union[IPv4, IPv6]],
+        destination_addresses: List[Union[IPv4, IPv6]],
+        icmp_code: Union[str, None],
+        icmp_type: Union[str, None],
+        source_interface: str,
+        source_ports: List[Union[Tuple[int, int], int]],
+        destination_ports: List[Union[Tuple[int, int], int]],
+        comment: List[str],
+        logging: List[str],
+        term_options: List[str],
+    ):
+        def to_network_addr(i: Union[IPv6, IPv4]):
+            return str(i.with_prefixlen)
+
+        options = [direction, action]
+
+        if protocol:
+            options.append("-proto %s" % self.PROXMOX_PROTO_MAP.get(protocol, protocol))
+
+        # proxmox firewall supports multiple sources/destinations per rule
+        if source_interface:
+            options.append("-iface %s" % source_interface)
+
+        # we cannot use address tokens since IPsets are defined in separate files (at cluster
+        # or at host level) and we're only outputting one file at a time, which may not be
+        # cluster/host firewall configuration.
+        if destination_addresses:
+            options.append("-dest %s" % ','.join(map(to_network_addr, destination_addresses)))
+
+        if source_addresses:
+            options.append("-source %s" % ','.join(map(to_network_addr, source_addresses)))
+
+        if source_ports and protocol in PROTOS_WITH_PORTS:
+            options.append(
+                "-sport %s" % ','.join(map(lambda p: str(ProxmoxPort(p)), source_ports))
+            )
+
+        if destination_ports and protocol in PROTOS_WITH_PORTS:
+            options.append(
+                "-dport %s" % ','.join(map(lambda p: str(ProxmoxPort(p)), destination_ports))
+            )
+
+        if protocol in ProxmoxIcmp.ICMP_PROTOS:
+            options.append('-icmp-type %s' % str(ProxmoxIcmp(protocol, icmp_type, icmp_code)))
+
+        if logging:
+            logging_key = 'true'
+            log_option = next(
+                map(
+                    lambda o: o if o in self._LOG_LEVELS_MAP.keys() else None,
+                    term_options or [None],
+                )
+            )
+            log = log_option or logging_key
+            options.append('-log %s' % self._LOG_LEVELS_MAP[log])
+
+        if comment:
+            options.append("# %s" % ' '.join(comment))
+
+        return ' '.join(options)
+
+
+class Proxmox(aclgenerator.ACLGenerator):
+    """Proxmox firewall policy object"""
+
+    # aerleon class props
+    _PLATFORM = 'proxmox'
+    SUFFIX = '.prxmxfw'
+    _TERM = Term
+    _LOG_LEVELS = list(_TERM.LOG_LEVELS_MAP_OPTIONS.values())
```

Comment:
Any reason some of the constants are defined in Term when they are only used in Proxmox?